drift/lib/src/runtime/api/connection_user.dart

```diff
@@ -489,6 +489,9 @@ abstract class DatabaseConnectionUser {
     }
 
     return await resolved.doWhenOpened((executor) {
+      if (interceptor != null) {
+        executor = executor.interceptWith(interceptor);
```

Comment:
I think we should only intercept the `transactionExecutor` here (so we need another extension on `TransactionExecutor` to apply interceptors that returns a `TransactionExecutor` too).

The parameter suggests that the interceptor will only be applied to the transaction, so that way feels more consistent.

Reply:
I thought about only intercepting the `transactionExecutor`, but that felt incomplete since it wouldn't invoke the `beginTransaction` callback, which is where the lifecycle of the transaction actually begins. This way the `begin/commit/rollbackTransaction` methods would be matched up, instead of leaving them unpaired (only the top-level one for nested).

In my case, I do need to initialize some things when the transaction begins, so that would require extra bookkeeping in the `runXxx` methods.

If you still think you would prefer to only intercept the `transactionExecutor`, then I would be happy to respect that choice and do it that way. Let me know.